### DISCUSSION TITLE
fieldArray not updated when nested fieldArray changes

### DISCUSSION THREAD

**Discussion by Raphaww** [react-hook-form/react-hook-form#6395] [Q&A]
I create a fieldArray with _useFieldArray_. It returns a 'fields' object:

` const parentArray = useFieldArray({name: "components", ...}) `
` const { fields} = parentArray `

and every field of this array has a fieldArray on its own (nested fieldArray)

` const childArray = useFieldArray({name: '${name}.children', ...}) `
` const { fields, append } = childArray `

when I append/remove a field inside this 'childArray'
` append({name: "foo"}) `

the 'parentArray' does not update:
`console.log(fieldArray);`
expected: 
`{ components: [ { children: [ { name: "foo" } ] } ]`
reality:
`{ components: [ { children: [ ] } ] }`

**codesandbox link**
https://codesandbox.io/s/late-cookies-8lxo9?file=/src/TextForm.js

**how to reproduce**
1. Open link, **on the left side is the form section and in the right is the info section**, notice how the form is already initialized and both, form and info, have the same data
<img width="1098" alt="Screen Shot 2021-08-30 at 17 26 00" src="https://user-images.githubusercontent.com/7801116/131414393-fcd5bba0-463a-4303-b354-aa5c9ed02826.png">

2. Click on 'Add Child' button, it is expected to add a form AND a info on the right
<img width="1098" alt="Screen Shot 2021-08-30 at 17 26 53" src="https://user-images.githubusercontent.com/7801116/131414496-2fb779b2-1d03-4d87-ad29-dcf7cb31c797.png">

3. The form has been added, but notice how the info was not updated
<img width="1094" alt="Screen Shot 2021-08-30 at 17 27 26" src="https://user-images.githubusercontent.com/7801116/131414545-d7dd07a7-0e9c-4320-9d57-e8df7e37197d.png">

4. If I click 'Add  component' (append children to 'parentArray')
<img width="1018" alt="Screen Shot 2021-08-30 at 17 36 53" src="https://user-images.githubusercontent.com/7801116/131415269-0fd084ee-1273-4bec-a665-edf1c8c2edb4.png">

6. the info section is now correctly updated
<img width="1024" alt="Screen Shot 2021-08-30 at 17 37 23" src="https://user-images.githubusercontent.com/7801116/131415278-a7146c0f-ea76-432b-9afe-8d255066140a.png">

This is probably by design but i was wondering if **something can be done to have this parentArray updated when a nested array is changed**. This example is just to show how the fieldArray object does not change, but in my project its a little bit more complicated than just show the data

**Comment by bluebill1049** [comment]
I think you will have to 

- watch/useWatch that field array child field array from neseted. 
- or turn the whole field array into fully controlled and value always flushed from the parent field array
